### PR TITLE
Fix dedicated server crash due to null sq context

### PR DESCRIPTION
**Pull request by wolf109909**
hi github

### CODE REVIEW

**Comment by wolf109909** [comment]
i fucked up a bit

**Comment by wolf109909** [comment]
why did i close it

**Comment by wolf109909** [comment]
why format fails (cries

**Comment by wolf109909** [comment]
it took me fucking 
30mins trying to fix clang format
i love myself

File: NorthstarDLL/hoststate.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include "r2engine.h"
 #include "limits.h"
 #include "squirrel.h"
-
+#include "dedicated.h"
```
Comment (GeckoEidechse) [inline NorthstarDLL/hoststate.cpp:11]:
```suggestion
#include "dedicated.h"

```

File: NorthstarDLL/hoststate.cpp [hunk 2]
```diff
@@ -105,11 +105,15 @@ void, __fastcall, (CHostState* self, double flCurrentTime, float flFrameTime))
 	}
 
 	// Run Squirrel message buffer
-	if (g_pSquirrel<ScriptContext::UI>->m_pSQVM != nullptr && g_pSquirrel<ScriptContext::UI>->m_pSQVM->sqvm != nullptr)
-		g_pSquirrel<ScriptContext::UI>->ProcessMessageBuffer();
+	// Don't run buffers that belongs to context that is not initialized on dedicated server
```
Comment (GeckoEidechse) [inline NorthstarDLL/hoststate.cpp:108]:
```suggestion
	// Don't run buffers that belong to context that is not initialized on dedicated server
```

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
Can't say much about the code due to lack of C++ knowledge but I can point out typos and formatting suggestions :>

**Review by emma-miler — APPROVED** [review]
While this PR does work, i think the underlying problem is that these contexts apparently do not get created properly on dedicated server, which i would consider to be a bug.

**Comment by GeckoEidechse** [comment]
@pg9182 wanna test this with Docker?

**Comment by emma-miler** [comment]
Superceded by #348

**Pull request closed without merging**